.mergify.yml

```diff
@@ -1,10 +0,0 @@
-pull_request_rules:
-  - name: automatic merge on CI success
-    conditions:
-      - check-success=ci/hercules/evaluation
-      - check-success=ci/hercules/derivations
-      - author=github-actions[bot]
-    actions:
-      merge:
-        method: merge
-      delete_head_branch: {}
```

Comment:
I didn't get why that is mutually exclusive with the terraform plan?

Reply:
If `terraform-deploy` is a github required status check for merging and it only runs on staging/trying branches mergify wouldn't be able to merge the PR as the status wouldn't be green. Same for restricting it to `terraform/**` path, non-terraform PRs also wouldn't be green.

```diff
diff --git a/.github/workflows/null.yml b/.github/workflows/null.yml
new file mode 100644
index 0000000..928d347
--- /dev/null
+++ b/.github/workflows/null.yml
@@ -0,0 +1,20 @@
+name: "null"
+
+on:
+  push:
+    branches:
+      - staging
+      - trying
+    paths-ignore:
+      - "terraform/**"
+  pull_request:
+    paths-ignore:
+      - "terraform/**"
+
+jobs:
+  terraform-deploy:
+    if: github.repository == 'nix-community/infra'
+    runs-on: ubuntu-latest
+    steps:
+      - name: skip
+        run: echo "no changed terraform files, skipping"
diff --git a/.github/workflows/terraform.yml b/.github/workflows/terraform.yml
index 3314741..268eded 100644
--- a/.github/workflows/terraform.yml
+++ b/.github/workflows/terraform.yml
@@ -5,6 +5,8 @@ on:
     branches:
       - staging
       - trying
+    paths:
+      - "terraform/**"
 
 jobs:
   terraform-deploy:

```

Could work around that problem with something like this that always sets a green status except for terraform PRs. However it wouldn't work for update flake PRs, as a PR from a github action can't trigger another action we'd need the update PRs to be submitted by a bot account.

Reply:
I guess for this repo it isn't really necessary that we make it a required check, only downside is that something could get merged by accident without being applied. Labelling the relevant PRs automatically would probably prevent that.